cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,310 @@
+    AtomicReference<List<StructuredRow>> results = new AtomicReference<>();
+    List<Field<?>> prefix = new ArrayList<>(3);
+    prefix.add(Fields.stringField(NAMESPACE_COL, namespace.getName()));
+    prefix.add(Fields.longField(GENERATION_COL, namespace.getGeneration()));
+    prefix.add(Fields.stringField(OWNER_COL, owner));
+
+    try {
+      transactionRunner.run(context -> {
+        StructuredTable table = context.getTable(TABLE_ID);
+        Range range = Range.singleton(prefix);
+        List<StructuredRow> temp = new ArrayList<>();
+        try (CloseableIterator<StructuredRow> rowIter = table.scan(range, Integer.MAX_VALUE)) {
+          while (rowIter.hasNext()) {
+            temp.add(rowIter.next());
+          }
+        }
+        results.set(temp);
+      });
+    } catch (TransactionException e) {
+      Throwables.propagate(e.getCause());
+    }
+
+    List<StructuredRow> sortedResults = doSort(results.get(), sortRequest);
+    return sortedResults.stream().map(this::fromRow).collect(Collectors.toList());
+  }
+
+  /**
+   * Helper method to apply the sorting onto a list of rows. Sorting needs to take place at the store-level so it can
+   * leverage the StructuredRow to enable sorting on any field.
+   *
+   * @param rows list of {@link StructuredRow} to be sorted
+   * @param sortRequest {@link SortRequest} describing the sort to be performed
+   * @return a sorted list of {@link StructuredRow}
+   */
+  private List<StructuredRow> doSort(List<StructuredRow> rows, SortRequest sortRequest) {
+    if (sortRequest == null) {
+      return rows;
+    }
+    String sortField = sortRequest.getFieldName();
+    Map<String, FieldType.Type> fieldTypes = TABLE_SPEC
+      .getFieldTypes()
+      .stream()
+      .collect(Collectors.toMap(FieldType::getName, FieldType::getType));
+    if (!fieldTypes.containsKey(sortField)) {
+      throw new IllegalArgumentException(
+        String
+          .format("Invalid value '%s' for sortBy. This field does not exist in the Drafts table.",
+                  sortField));
+    }
+
+    rows.sort((o1, o2) -> {
+      FieldType.Type fieldType = fieldTypes.get(sortField);
+      Comparator<StructuredRow> comparator = null;
+      switch (fieldType) {
+        case STRING:
+          comparator = Comparator.<StructuredRow, String>comparing(o -> o.getString(sortField));
+          break;
+        case INTEGER:
+          comparator = Comparator.<StructuredRow, Integer>comparing(o -> o.getInteger(sortField));
+          break;
+        case LONG:
+          comparator = Comparator.<StructuredRow, Long>comparing(o -> o.getLong(sortField));
+          break;
+        case FLOAT:
+          comparator = Comparator.<StructuredRow, Float>comparing(o -> o.getFloat(sortField));
+          break;
+        case DOUBLE:
+          comparator = Comparator.<StructuredRow, Double>comparing(o -> o.getDouble(sortField));
+          break;
+        case BYTES:
+          throw new IllegalArgumentException(
```

Comment:
You can use the `Bytes.BYTES_COMPARATOR`

Reply:
done